File: DataStruct/strings/anagram.cpp
#include<stdio.h>
#include<iostream>

using namespace std;

bool anagram(string s1, string s2)

{
if (s1.length != s2.length)
    return false;
int count[256]={0};
for (int i=0; i<s1.length; i++){count[s1]++;}
for (int i=0; i<s2.length; i++){count[s2]--;}
for (int i=0; i<256; i++){if count!=0 {return false;}}
return true;
}

File: DataStruct/strings/anagram.py
s1 = 'asdfg'
s2 = 'gadsdfa'
s1 = [char for char in s1]
s2 = [char for char in s2]

def anagram(s1, s2):
    if len(s1) != len(s2):
        return False

    count = [0]*256
    for i in range(0,len(s1)):
        count[i]=count[i]+1

    for i in range(0,len(s2)):
        count[i]-=1

    for i in range(256):
        if count[i] != 0:
            return False

    return True

print(anagram(s1,s2))